### ISSUE TITLE
Subversion 1.8 fails to build

### ISSUE THREAD

**Issue by aik099** [Homebrew/homebrew-core#44226]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```
→ brew config
HOMEBREW_VERSION: 2.1.11
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 37714b5ce1e4909d4d61de2af98343c7012f7cd9
Last commit: 3 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 15f5d77d1bd788c21f9a95e9061b9e8250f63022
Core tap last commit: 2 days ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_GITHUB_API_TOKEN: set
CPU: octa-core 64-bit ivybridge
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1001
Git: 2.23.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_74, 1.7.0_60
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: N/A
CLT headers: 10.3.0.0.1.1562985497

→ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.bash_profile
```

To help us debug your issue please explain:

- What you were trying to do (and why)

I'm installing Subversion 1.8 (not the latest one) using `brew install subversion@1.8` command.

- What happened (include command output)

```
→ brew install subversion@1.8
==> Downloading https://homebrew.bintray.com/bottles/subversion@1.8-1.8.19_1.mojave.bottle.1.tar.gz
Already downloaded: /Users/alex/Library/Caches/Homebrew/downloads/889686ec20e8bdbdee564c5d86835a299696daf161f76a9755eb208d5551c510--subversion@1.8-1.8.19_1.mojave.bottle.1.tar.gz
==> Pouring subversion@1.8-1.8.19_1.mojave.bottle.1.tar.gz
==> Caveats
svntools have been installed to:
  /usr/local/opt/subversion@1.8/libexec

subversion@1.8 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

If you need to have subversion@1.8 first in your PATH run:
  echo 'export PATH="/usr/local/opt/subversion@1.8/bin:$PATH"' >> ~/.bash_profile

For compilers to find subversion@1.8 you may need to set:
  export LDFLAGS="-L/usr/local/opt/subversion@1.8/lib"
  export CPPFLAGS="-I/usr/local/opt/subversion@1.8/include"


Bash completion has been installed to:
  /usr/local/opt/subversion@1.8/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/subversion@1.8/1.8.19_1: 130 files, 9.2MB

→ /usr/local/Cellar/subversion@1.8/1.8.19_1/bin/svn --version

dyld: Library not loaded: /usr/local/opt/libmagic/lib/libmagic.1.dylib
  Referenced from: /usr/local/bin/svn
  Reason: image not found
Abort trap: 6
```

- What you expected to happen

After installing I've expected subversion to be in a working state with all dependencies installed, but `libmagic` even though being used by subversion wasn't listed as a dependency in the formulae.

Doing `brew install libmagic` separately fxed the issue.

I had subversion@1.8 installed long time ago and it worked perfectly. After recent `brew upgrade` command run it became broken.

- Step-by-step reproduction instructions (by running `brew install` commands)

See above.

**Comment by fxcoudert** [comment]
This should be fixed with 84c9378ef9d8b67984f2cc9aad8570c2581a0229

**Comment by aik099** [comment]
Doing `brew up` doesn't show up `subversion@1.8` as updated formulae.

How exactly hash changing (done in above shown commit) have fixed an issue? I'm not seeing `libmagic` added as a dependency either.
